### PR TITLE
Add chapter 22.

### PR DESCRIPTION
**Pull request by OrangeToucan**

### CODE REVIEW

File: docs/22-storage-management.markdown [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+to being compressed in size by the long-term growth of other spaces.
+
+Internal storage has both pure and impure parts. The interpreter
+program itself is pure and sharable, while impure storage is used for
+internal pointers, counters, and flags, for example, pointers to the
+boundaries of other spaces. In the pure part of this space are most
+of the `ATOM`s in an initial MDL, along with their `OBLIST` buckets
+(`LIST`s) and `GVAL` slots (a pure extension of the global vector),
+where possible. A `SET` or `SETG` of a pure `ATOM` automatically
+impurifies the `ATOM` and as much of its `OBSLIST` bucket as needs to
```
Comment (taradinoc) [inline docs/22-storage-management.markdown]: `OBSLIST` -> `OBLIST`

File: docs/22-storage-management.markdown [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+creates an inferior operating-system process (named `AGC` in the ITS
+version) whose address space is used to hold the new copies of
+non-garbage objects. MDL accesses the inferior's address space
+through two pages ("frontier" and "window") in its internal space
+that are shared with the inferior. If the garbage collection occurred
+because movable garbage-collected space was exhausted, then the
+"mark-sweep" algorithm might be used instead (see below) and no
+inferior process is created.
+3. The garbage collector marks and moves all objects that can
+possibly be referenced hereafter. It begins with the <MAIN> PROCESS
```
Comment (taradinoc) [inline docs/22-storage-management.markdown]: backtick (separately) `<MAIN>` `PROCESS`

File: docs/22-storage-management.markdown [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+and the currently running PROCESS <ME>, considered as vectors
```
Comment (taradinoc) [inline docs/22-storage-management.markdown]: ...and `PROCESS` `<ME>`

File: docs/22-storage-management.markdown [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+containing the control stacks, object pointers in live registers,
+etc. Every object in these "PROCESS vectors" is marked "accessible",
```
Comment (taradinoc) [inline docs/22-storage-management.markdown]: backtick `PROCESS`

File: docs/22-storage-management.markdown [hunk 5]
```diff
@@ -0,0 +1,371 @@
...
+and every element of these objects (bindings, etc.), and so on
+recursively. The "copying" algorithm moves objects into the inferior
+processes' address space as it marks them.
```
Comment (taradinoc) [inline docs/22-storage-management.markdown]: processes' -> process's

File: docs/22-storage-management.markdown [hunk 6]
```diff
@@ -0,0 +1,371 @@
...
+4. If the garbage collection is "exhaustive" – which is possible only
```
Comment (taradinoc) [inline docs/22-storage-management.markdown]: dash style: `--` (also in the next line)

File: docs/22-storage-management.markdown [hunk 7]
```diff
@@ -0,0 +1,371 @@
...
+in the "copying" algorithm – then both the chain of associations and
+top-level local/global bindings are examined thoroughly, which takes
+more time but is more likely to uncover garbage therein. In a normal
+garbage collection these constructs are not treated specially.
+5. Finally, the "mark-sweep" algorithm sweeps through the storage
+space, adding unmarked objects to the internal free lists for later
+re-use. The "copying" algorithm maps the inferior process's address
+space into MDL's own, replace garbagey storage with the new compact
```
Comment (taradinoc) [inline docs/22-storage-management.markdown]: replace garbagey -> replacing old garbagey

File: docs/22-storage-management.markdown [hunk 8]
```diff
@@ -0,0 +1,371 @@
...
+storage, and the inferior is destroyed.
```
Comment (taradinoc) [inline docs/22-storage-management.markdown]: inferior -> inferior process

File: docs/22-storage-management.markdown [hunk 9]
```diff
@@ -0,0 +1,371 @@
...
+
+## 22.5 GC
+
+    <GC min:fix exh?:false-or-any>
```
Comment (taradinoc) [inline docs/22-storage-management.markdown]: third parameter: ms-freq:fix

File: docs/22-storage-management.markdown [hunk 10]
```diff
@@ -0,0 +1,371 @@
...
+for use) movable garbage-collected storage the garbage collector will
+be satisfied with having after it is done. Initially it is 8192
+words. If the total amount of reclaimed storage is less than *min*,
+the garbage collector will ask the operating system for enough
+storage (in 1024 word blocks) to make it up. N.B.: the system may be
+incivil enough not to grant the request; in that case, the garbage
+collector will be content with what it has, **unless** that is not
+enough to satisfy a **pending** request for storage. Then it will
+inform you that it is losing. A large *min* will result in fewer
+total garbage collections, but that will take longer since the total
```
Comment (taradinoc) [inline docs/22-storage-management.markdown]: that -> they

File: docs/22-storage-management.markdown [hunk 11]
```diff
@@ -0,0 +1,371 @@
...
+
+## 22.6. BLOAT
+
+`BLOAT` is used to cause a temporary expansion of the available
+storage space with or without changing the garbage-collection
+parameters. `BLOAT` is particularly useful for avoiding unnecessary
+garbage collections when loading a large file. It will cause (at
+most) one garbage collection, at the end of which the available
+storage will be at least the amount specified in the call to `BLOAT`.
+(Unless. of course, the operating system is cranky and will not
```
Comment (taradinoc) [inline docs/22-storage-management.markdown]: period -> comma

File: docs/22-storage-management.markdown [hunk 12]
```diff
@@ -0,0 +1,371 @@
...
+provide the storage. Then you will get an error. `<ERRET 1>` from
+this error will cause the `BLOAT` to return `1`, which usually just
+causes you to lose at a later time - unless the operating system
```
Comment (taradinoc) [inline docs/22-storage-management.markdown]: dash style: `--`

File: docs/22-storage-management.markdown [hunk 13]
```diff
@@ -0,0 +1,371 @@
...
+feels nicer when the storage is absolutely necessary.)
+
+A call to BLOAT looks like this:
+
+    <BLOAT fre stk lcl glb typ sto pstk
+           min plcl pglb pytp imp pur dpstk dstk>
```
Comment (taradinoc) [inline docs/22-storage-management.markdown]: pytp -> ptyp

File: docs/22-storage-management.markdown [hunk 14]
```diff
@@ -0,0 +1,371 @@
...
+
+where all all arguments on the first line above are `FIX`, optional
```
Comment (taradinoc) [inline docs/22-storage-management.markdown]: duplicated "all"

File: docs/22-storage-management.markdown [hunk 15]
```diff
@@ -0,0 +1,371 @@
...
+20. number of slots for `TYPE`s in `TYPE` vector
+21. number of `TYPE`s existing
+22. number of words of immovable garbage-collected storage
+23. number of words of immovable storage unused
+24. size of largest unused contiguous immovable-storage block
+25. number of words on internal stack
+26. number of words on internal stack in use
+27. maximum size of internal stack ever reached
+
+## 22.8. GC-HON
```
Comment (taradinoc) [inline docs/22-storage-management.markdown]: GC-MON

File: docs/22-storage-management.markdown [hunk 16]
```diff
@@ -0,0 +1,371 @@
...
+
+    <GC-MOND pred>
```
Comment (taradinoc) [inline docs/22-storage-management.markdown]: GC-MON

File: docs/22-storage-management.markdown [hunk 17]
```diff
@@ -0,0 +1,371 @@
...
+
+("garbage-collector monitor") determines whether or not the
+interpreter will hereafter print information on the console when a
```
Comment (taradinoc) [inline docs/22-storage-management.markdown]: console -> terminal

File: docs/22-storage-management.markdown [hunk 18]
```diff
@@ -0,0 +1,371 @@
...
+
+returns *old*, after causing a miniature garbage collection to occur,
+during which **all** references to *old* are changed so as to refer
+to *new*. Neither argument can be of `PRIMTYPE` `STRING` or `BYTES`
+or `LOCD` or live on the control stack, unless both are of the same
+`PRIMTYPE`. One `TYPE` name cannot be substituted for another. One of
+the few legitimate uses for it is to substitute the "right" `ATOM`
+for the "wrong" one, after `OBLIST`s have been in the wrong state.
+This is more or less the way `ATOM`s are impurified. It is also
+useful for unlinking `RSUBR`s. `SUBSTITUTE` returns old as a favor:
```
Comment (taradinoc) [inline docs/22-storage-management.markdown]: italic *old* (also in the next line)

File: docs/22-storage-management.markdown [hunk 19]
```diff
@@ -0,0 +1,371 @@
...
+unless you hang onto old at that point, it will be garbage-collected.
+
+## 22.9.2 PURIFY
+
+    <PURIFY any-1 ... any-n>
```
Comment (taradinoc) [inline docs/22-storage-management.markdown]: any-n -> any-N

File: docs/22-storage-management.markdown [hunk 20]
```diff
@@ -0,0 +1,371 @@
...
+
+returns its last argument, after causing a miniature garbage
+collection that results in all the arguments becoming pure and
+sharable, and ignored afterward by the garbage collector. No argument
+can live on the control stack or be of `PRIMTYPE` `PROCESS` or `LOCD`
+or `ASOC`. Sharing between jobs or forks actually occurs after a
```
Comment (taradinoc) [inline docs/22-storage-management.markdown]: jobs or forks -> operating-system processes

**Comment by OrangeToucan** [comment]
Hello, I have made these changes.

**Pull request merged**